sensor/isl29125/isl29125.go

```diff
@@ -0,0 +1,242 @@
+// Package isl29125 allows interfacing with the ISL29125 RGB light sensor
+// Datasheet: http://www.intersil.com/content/dam/Intersil/documents/isl2/isl29125.pdf
+package isl29125
+
+/*
+
+   TODO:
+       - add support for the following:
+           - config register 2 and 3
+           - lux calculation
+           - powerdown / power up
+           - interupt config / monitor
+*/
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kidoman/embd"
+)
+
+const (
+	pollDelay = 250
+)
+
+const (
+	ModePowerDown = 0x00
+	ModeGreenOnly = 0x01
+	ModeRedOnly   = 0x02
+	ModeBlueOnly  = 0x03
+	ModeStandBy   = 0x04
+	ModeRGB       = 0x05
+	ModeRG        = 0x06
+	ModeGB        = 0x07
+)
+
+const (
+	LuxRange375 = 0x00
+	LuxRange10k = 0x08
+)
+
+const (
+	IRAdjustLow  = 0x00
+	IRAdjustMed  = 0x20
+	IRAdjustHigh = 0x3f
+)
+
+const (
+	Resolution16Bit = 0x00
+	Resolution12Bit = 0x10
+)
+
+const (
+	SyncStartOnWrite = 0x00
+	SyncStartOnInt   = 0x20
+)
+
+const (
+	RegisterDeviceID  = 0x00
+	RegisterConfig1   = 0x01
+	RegisterConfig2   = 0x02
+	RegisterConfig3   = 0x03
+	RegisterFlags     = 0x08
+	RegisterGreenLow  = 0x09
+	RegisterGreenHigh = 0x0a
+	RegisterRedLow    = 0x0b
+	RegisterRedHigh   = 0x0c
+	RegisterBlueLow   = 0x0d
+	RegisterBlueHigh  = 0x0e
+)
+
+const (
+	FlagReady               = 0x00
+	FlagInterrupt           = 0x01
+	FlagConversion          = 0x02
+	FlagPowerDownOrBrownOut = 0x04
+	FlagGreenConverting     = 0x10
+	FlagRedConverting       = 0x20
+	FlagBlueConverting      = 0x30
+)
+
+const (
+	DefaultConfig = ModeRGB | LuxRange375 | Resolution16Bit | SyncStartOnWrite
+)
+
+const (
+	SensorAddr = 0x44
+	DeviceID   = 0x7d
+)
+
+const (
+	CmdGetStatus = 0x08
+	CmdReset     = 0x46
+)
+
+// Reading represents a single reading from an RGB light sensor
+type Reading struct {
+	Red        uint16
+	Green      uint16
+	Blue       uint16
+	Lux        uint16
+	LuxRange   int
+	Resolution int
+}
+
+// ISL29125 represents an RGB light sensor
+type ISL29125 struct {
+	Bus  embd.I2CBus
+	Poll int
+
+	readings chan *Reading
+	quit     chan bool
+
+	mode uint8
+}
+
+// New returns an ISL29125 for a given config
+func New(config uint8, bus embd.I2CBus) *ISL29125 {
+	glog.Info("Creating new ISL29125")
+	return &ISL29125{Bus: bus, Poll: pollDelay, mode: config}
+}
+
+func (i *ISL29125) Init() error {
```

Comment:
Realized this PR was still out there and going to clean it up. Can you be more clear. When you say 'This' are you referring to the presence of the Init method, the inclusion of a config param to New or the order of the params to New?